pywr-schema/src/metric.rs

```diff
@@ -401,3 +463,143 @@ impl EdgeReference {
+        match self {
+            Self::Node(node_ref) => node_ref.load_usize(network, args),
+            // Global parameter with no parent
+            Self::Parameter(parameter_ref) => parameter_ref.load_usize(network, None),
+            // Local parameter loaded from parent's namespace
+            Self::LocalParameter(parameter_ref) => {
+                if parent.is_none() {
+                    return Err(SchemaError::LocalParameterReferenceRequiresParent(
+                        parameter_ref.name.clone(),
+                    ));
+                }
+
+                parameter_ref.load_usize(network, parent)
+            }
+            Self::Constant { value } => Ok((*value).into()),
+            Self::Table(table_ref) => {
+                let value = args
+                    .tables
+                    .get_scalar_usize(table_ref)
+                    .map_err(|error| SchemaError::TableRefLoad {
+                        table_ref: table_ref.clone(),
+                        error,
+                    })?;
+                Ok(value.into())
+            }
+            Self::Timeseries(ts_ref) => {
+                let param_idx = match &ts_ref.columns {
+                    Some(TimeseriesColumns::Scenario(scenario)) => {
+                        args.timeseries
+                            .load_df_usize(network, ts_ref.name.as_ref(), args.domain, scenario.as_str())?
+                    }
+                    Some(TimeseriesColumns::Column(col)) => {
+                        args.timeseries
+                            .load_column_usize(network, ts_ref.name.as_ref(), col.as_str())?
+                    }
+                    None => args
+                        .timeseries
+                        .load_single_column_usize(network, ts_ref.name.as_ref())?,
+                };
+                Ok(param_idx.into())
+            }
+            Self::InterNetworkTransfer { name } => {
+                // Find the matching inter model transfer
+                match args.inter_network_transfers.iter().position(|t| &t.name == name) {
+                    Some(idx) => Ok(MetricUsize::InterNetworkTransfer(MultiNetworkTransferIndex(idx))),
+                    None => Err(SchemaError::InterNetworkTransferNotFound(name.to_string())),
+                }
+            }
+        }
+    }
+}
+
+impl TryFromV1<ParameterValueV1> for IndexMetric {
+    type Error = ConversionError;
+
+    fn try_from_v1(
+        v1: ParameterValueV1,
+        parent_node: Option<&str>,
+        conversion_data: &mut ConversionData,
+    ) -> Result<Self, Self::Error> {
+        let p = match v1 {
+            // There was no such thing as s constant index in Pywr v1
+            // TODO this could print a warning and do a cast to usize instead.
+            ParameterValueV1::Constant(value) => {
+                // Check if the value is not a whole non-negative number
+                if value.fract() != 0.0 && value >= 0.0 {
```

Comment:
Should this be `value.fract() != 0.0 || value < 0.0`

Reply:
Fixed and added a test for it.